src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_global_properties_request.cc

```diff
@@ -642,7 +653,24 @@ void SetGlobalPropertiesRequest::PrepareUIRequestMenuAndKeyboardData(
   if (is_keyboard_props_present) {
     out_params[hmi_request::keyboard_properties] =
         msg_params[hmi_request::keyboard_properties];
-    app->set_keyboard_props(msg_params[hmi_request::keyboard_properties]);
+
+    if (msg_params[hmi_request::keyboard_properties].keyExists(
+            hmi_request::keyboard_layout)) {
+      app->set_keyboard_props(msg_params[hmi_request::keyboard_properties]);
+      return;
+    }
+
+    smart_objects::SmartObject cached_keyboard_props(
+        msg_params[hmi_request::keyboard_properties]);
```

Comment:
Seems to me that the `language` would also be cached. Possibly `keypressMode` and `autoCompleteList` as well.

Reply:
@jacobkeeler Caching of keyboard capabilities is improved in 8627500
Now `language` and `autoCompleteList` are cached too (caching of keypressMode is not needed, because handling of this parameter is ensured from HMI side, so there is no need in additional actions from SDL side)